careers/careers/views.py

```diff
@@ -1,13 +1,40 @@
 from django.views.generic import DetailView, ListView, TemplateView
 from django.shortcuts import get_object_or_404
+from jinja2 import Markup
 
 from careers.careers.forms import PositionFilterForm
 from careers.careers.models import Position
+from careers.careers.wordpress import complete_posts_data, get_posts_data, process_excerpt
 
 
 class HomeView(TemplateView):
     template_name = 'careers/home.jinja'
 
+    def get_context_data(self, **kwargs):
+        context = super(HomeView, self).get_context_data(**kwargs)
+
+        # blog content isn't critical to the page. if the wordpress API fails
+        # for some reason, we can just move on and wait for the next build
+        try:
+            posts_data = get_posts_data()
+            blog_posts = []
+
+            if posts_data:
+                complete_posts_data(posts_data)
+
+                for post in posts_data:
+                    blog_posts.append({
+                        'title': Markup(post['title']['rendered']).unescape(),
+                        'link': post['link'],
+                        'excerpt': Markup(process_excerpt(post['excerpt']['rendered'])),
+                        'image': post['featured_media']['media_details']['sizes']['post-large']['source_url'],  # noqa
+                    })
+        except Exception:
```

Comment:
This is a overly generic exception which can bite us in the future. For example if blog.m.o API endpoint changes or the schema changes it will be long before we actually figure out that we don't display any blog posts on the page. 

I see two possible solutions:
 - We allow this to fail, we set the GitLab job to get retried a couple of times if it fails. If it continues to fail when at most in 24 hours we will a DMS alert that the page hasn't been rebuilt. We know that something is wrong
 - We keep the exception as is and we create a new DMS snitch that gets triggered only when we successfully get blog posts. If the alert goes off we know we need to look at this functionality but the site keeps getting new job listings and everything operates as usual. Alternatively we could log to Sentry.

I prefer the later solution and while I won't block on this now due to the deadlines, I highly suggest that we implement something here in the near future. Given our current record on careers I don't see us working on it for the next 5 years, so I want to have a robust solution.

Reply:
I like that it means that a lack of blog posts won't block updates to the job listings. But +1 for some DMS monitoring.